src/notice/mod.rs

```diff
@@ -0,0 +1,140 @@
+//!
+//! #[derive(Debug)]
+//! struct MyError;
+//! impl Error for MyError {}
+//! impl Display for MyError {
+//!     fn fmt(&self, f: &mut Formatter<'_>) -> Result { write!(f, "") }
+//! }
+//! let my_error = MyError {};
+//!
+//! let ne: NoticeError = my_error.into();
+//! ```
+//!
+//! Typically you won't need to work with the NoticeError directly, since you
+//! can add errors to a Notice using the `.add_error` function.
+//!
+//! ```
+//! use std::error::Error;
+//! use std::fmt::{Display, Formatter, Result};
+//! use airbrake::{Notice, NoticeError};
+//!
+//! #[derive(Debug)]
+//! struct MyError;
+//! impl Error for MyError {}
+//! impl Display for MyError {
+//!     fn fmt(&self, f: &mut Formatter<'_>) -> Result { write!(f, "") }
+//! }
+//! let my_error = MyError {};
+//!
+//! let notice = Notice::builder()
+//!     .add_error(my_error)
+//!     .build();
+//! ```
+//!
+//! Backtraces are also supported in error reporting. You will typically have
+//! a raw backtrace created by the backtrace crate. These sort of backtraces
+//! can be added using the `raw_backtrace` builder method.
+//! ```
+//! use airbrake::{Notice, NoticeError};
+//! use airbrake::backtrace::Backtrace;
+//!
+//! let notice_error = NoticeError::builder("foo")
+//!     .raw_backtrace(&Backtrace::new())
+//!     .build();
+//! let notice = Notice::builder()
+//!     .add_notice(notice_error)
+//!     .build();
+//! ```
+//!
+//! You can also include a backtrace while building a Notice based on an
+//! error, however since backtraces aren't part of the Error trait, you'll
+//! have to provide it using the add_error_with_backtrace() function:
+//!
+//! ```
+//! use std::error::Error;
+//! use std::fmt::{Display, Formatter, Result};
+//! use backtrace;
+//! use airbrake::{Notice, NoticeError};
+//! use airbrake::backtrace::Backtrace;
+//!
+//! #[derive(Debug)]
+//! struct MyError;
+//! impl Error for MyError {}
+//! impl Display for MyError {
+//!     fn fmt(&self, f: &mut Formatter<'_>) -> Result { write!(f, "") }
+//! }
+//! let my_error = MyError {};
+//! let my_backtrace = Backtrace::new();
+//!
+//! let notice = Notice::builder()
+//!     .add_error_with_backtrace(my_error, my_backtrace)
+//!     .build();
+//! ```
+//!
+//! Airbrake supports multiple errors being logged in a single notification,
+//! so using `.add_error` and `.add_notice` will append to the list of errors
+//! that contained. If you have multiple errors ready, you can add them all
+//! at once using `.add_errors` or `.add_notices`, which accept iterators.
+//!
+//! ```
+//! use std::error::Error;
+//! use airbrake::{Notice, NoticeError};
+//!
+//! let my_error1 = NoticeError::new("foo", None, None);
+//! let my_error2 = NoticeError::new("bar", None, None);
+//! let error_list = vec![my_error1, my_error2].into_iter();
+//! let notice = Notice::builder()
+//!     .add_notices(error_list)
+//!     .build();
+//! ```
+//!
+//! The Context struct represents the context the service is running in, like
```

Comment:
Some parts of this docstring are duplicates of the docstring added to the main `lib.rs` file. Doc repetition should be kept to a minimal, but this is probably something that can be sorted out in sprint 2, as docs are going to need more thorough organization anyway